crates/rpc/src/methods/token_add.rs

```diff
@@ -0,0 +1,317 @@
+        let current_network = networks.get_current();
+        Ok(current_network.chain_id)
+    }
+
+    pub async fn get_current_wallet_address(&self) -> Result<Address> {
+        let wallets = Wallets::read().await;
+        let wallet_address = wallets.get_current_wallet().get_current_address().await;
+        Ok(wallet_address)
+    }
+
+    pub fn check_type(&self) -> Result<()> {
+        if self._type != "ERC20" && self._type != "ERC721" && self._type != "ERC1155" {
+            return Err(Error::TypeInvalid(self._type.clone()));
+        }
+
+        Ok(())
+    }
+
+    pub fn check_symbol(&self) -> Result<()> {
+        if let Some(erc20_token) = &self.erc20_token {
+            let symbol = erc20_token.symbol.clone().unwrap_or_default();
+            if symbol.len() > 11 {
+                return Err(Error::SymbolInvalid(symbol));
+            }
+        }
+        Ok(())
+    }
+
+    pub fn check_decimals(&self) -> Result<()> {
+        if let Some(erc20_token) = &self.erc20_token {
+            let decimals = erc20_token.decimals.unwrap_or(0);
+            if decimals > 36 {
+                return Err(Error::DecimalsInvalid(decimals));
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn check_network(&self) -> Result<()> {
+        let current_chain_id = self.get_current_chain_id().await?;
+        let chain_id = self.chain_id.unwrap_or(current_chain_id);
+        if current_chain_id != chain_id {
+            return Err(Error::NetworkInvalid);
+        }
+        Ok(())
+    }
+
+    pub async fn run(self) -> Result<()> {
+        self.check_type()?;
+        self.check_network().await?;
+
+        match self._type.as_str() {
+            "ERC20" => {
+                self.check_symbol()?;
+                self.check_decimals()?;
+                // TODO: check if 'type' from call matches the added 'tokenType'
+            }
+            "ERC721" => {
+                // TODO: check if 'type' from call matches the added 'tokenType'
+                // TODO: check if wallet is the owner of the token
+            }
+            "ERC1155" => {
+                // TODO: check if 'type' from call matches the added 'tokenType'
+                // TODO: check if wallet is the owner of the token
+            }
+            _ => return Err(Error::TypeInvalid(self._type.clone())),
+        }
+
+        let dialog = match self._type.as_str() {
+            "ERC20" => Dialog::new(
+                "erc20token-add",
+                serde_json::to_value(&self.erc20_token).unwrap(),
+            ),
+            "ERC721" => Dialog::new(
+                "erc721token-add",
+                serde_json::to_value(&self.erc721_token).unwrap(),
+            ),
+            "ERC1155" => Dialog::new(
+                "erc1155token-add",
+                serde_json::to_value(&self.erc1155_token).unwrap(),
+            ),
+            _ => return Err(Error::TypeInvalid(self._type.clone())),
+        };
```

Comment:
why not merge these two `match` statements since they check the exact same thing?